src/date-range-picker/calendar/index.tsx

```diff
@@ -138,59 +121,54 @@ export default function DateRangePickerCalendar({
     };
 
     // If both fields are empty, we set the start date
-    if (!rangeStart.dateString && !rangeEnd.dateString) {
+    if (!start.date && !end.date) {
       const startDate = startOfDay(selectedDate);
-      rangeStart.setDate(startDate);
+      onRangeChange({ date: formatDate(startDate), time: '' }, undefined);
```

Comment:
As I figured from manual testing, when changing any date, we discard any time modifications